### ISSUE TITLE
Test failures after pip installation

### ISSUE THREAD

**Issue by johndeppe** [stanford-mast/nn_dataflow#17]
Hi Professor Gao,

I'm just getting started with nn_dataflow and I'm failing a couple tests after installing. I've attached my terminal contents. Do you have any suggestions? I'm using Ubuntu 18.04, if it matters.

Thank you.

```
deppe@bardiche:~$ git clone https://github.com/stanford-mast/nn_dataflow.git
Cloning into 'nn_dataflow'...
remote: Enumerating objects: 300, done.
remote: Counting objects: 100% (300/300), done.
remote: Compressing objects: 100% (212/212), done.
remote: Total 3885 (delta 171), reused 160 (delta 88), pack-reused 3585
Receiving objects: 100% (3885/3885), 1.05 MiB | 3.14 MiB/s, done.
Resolving deltas: 100% (2994/2994), done.
deppe@bardiche:~$ cd nn_dataflow
deppe@bardiche:~/nn_dataflow$ pip3 install --user -e .
Obtaining file:///home/deppe/nn_dataflow
Collecting argparse (from nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/f2/94/3af39d34be01a24a6e65433d19e107099374224905f1e0cc6bbe1fd22a2f/argparse-1.4.0-py2.py3-none-any.whl
Collecting coverage>=4 (from nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/2a/3e/fc18ecef69f174c13493576f46966053c1da07fd8721962530dc1a10b1ca/coverage-5.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting fastcache>=1 (from nn-dataflow==2.1)
Collecting pytest-cov>=2 (from nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/b9/54/3673ee8be482f81527678ac894276223b9814bb7262e4f730469bb7bf70e/pytest_cov-2.8.1-py2.py3-none-any.whl
Collecting pytest-xdist>=1 (from nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/7c/8c/7f93c1d82f25a69a1c6e68189b9cf5ddce08dcaefdbd913d328b0234e13b/pytest_xdist-1.31.0-py2.py3-none-any.whl
Collecting pytest>=3 (from nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/c7/e2/c19c667f42f72716a7d03e8dd4d6f63f47d39feadd44cc1ee7ca3089862c/pytest-5.4.1-py3-none-any.whl
Collecting sympy>=1 (from nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/ce/5b/acc12e3c0d0be685601fc2b2d20ed18dc0bf461380e763afc9d0a548deb0/sympy-1.5.1-py2.py3-none-any.whl
Collecting pytest-forked (from pytest-xdist>=1->nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/03/1e/81235e1fcfed57a4e679d34794d60c01a1e9a29ef5b9844d797716111d80/pytest_forked-1.1.3-py2.py3-none-any.whl
Requirement already satisfied: six in /usr/lib/python3/dist-packages (from pytest-xdist>=1->nn-dataflow==2.1) (1.11.0)
Collecting execnet>=1.1 (from pytest-xdist>=1->nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/d3/2e/c63af07fa471e0a02d05793c7a56a9f7d274a8489442a5dc4fb3b2b3c705/execnet-1.7.1-py2.py3-none-any.whl
Collecting packaging (from pytest>=3->nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/62/0a/34641d2bf5c917c96db0ded85ae4da25b6cd922d6b794648d4e7e07c88e5/packaging-20.3-py2.py3-none-any.whl
Collecting importlib-metadata>=0.12; python_version < "3.8" (from pytest>=3->nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/ad/e4/891bfcaf868ccabc619942f27940c77a8a4b45fd8367098955bb7e152fb1/importlib_metadata-1.6.0-py2.py3-none-any.whl
Collecting more-itertools>=4.0.0 (from pytest>=3->nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/72/96/4297306cc270eef1e3461da034a3bebe7c84eff052326b130824e98fc3fb/more_itertools-8.2.0-py3-none-any.whl
Collecting py>=1.5.0 (from pytest>=3->nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/99/8d/21e1767c009211a62a8e3067280bfce76e89c9f876180308515942304d2d/py-1.8.1-py2.py3-none-any.whl
Requirement already satisfied: wcwidth in /usr/lib/python3/dist-packages (from pytest>=3->nn-dataflow==2.1) (0.1.7)
Collecting attrs>=17.4.0 (from pytest>=3->nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/a2/db/4313ab3be961f7a763066401fb77f7748373b6094076ae2bda2806988af6/attrs-19.3.0-py2.py3-none-any.whl
Collecting pluggy<1.0,>=0.12 (from pytest>=3->nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/a0/28/85c7aa31b80d150b772fbe4a229487bc6644da9ccb7e427dd8cc60cb8a62/pluggy-0.13.1-py2.py3-none-any.whl
Collecting mpmath>=0.19 (from sympy>=1->nn-dataflow==2.1)
Collecting apipkg>=1.4 (from execnet>=1.1->pytest-xdist>=1->nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/67/08/4815a09603fc800209431bec5b8bd2acf2f95abdfb558a44a42507fb94da/apipkg-1.5-py2.py3-none-any.whl
Requirement already satisfied: pyparsing>=2.0.2 in /usr/lib/python3/dist-packages (from packaging->pytest>=3->nn-dataflow==2.1) (2.2.0)
Collecting zipp>=0.5 (from importlib-metadata>=0.12; python_version < "3.8"->pytest>=3->nn-dataflow==2.1)
  Using cached https://files.pythonhosted.org/packages/b2/34/bfcb43cc0ba81f527bc4f40ef41ba2ff4080e047acb0586b56b3d017ace4/zipp-3.1.0-py3-none-any.whl
Installing collected packages: argparse, coverage, fastcache, packaging, zipp, importlib-metadata, more-itertools, py, attrs, pluggy, pytest, pytest-cov, pytest-forked, apipkg, execnet, pytest-xdist, mpmath, sympy, nn-dataflow
  Running setup.py develop for nn-dataflow
Successfully installed apipkg-1.5 argparse-1.4.0 attrs-19.3.0 coverage-5.1 execnet-1.7.1 fastcache-1.1.0 importlib-metadata-1.6.0 more-itertools-8.2.0 mpmath-1.1.0 nn-dataflow packaging-20.3 pluggy-0.13.1 py-1.8.1 pytest-5.4.1 pytest-cov-2.8.1 pytest-forked-1.1.3 pytest-xdist-1.31.0 sympy-1.5.1 zipp-3.1.0
WARNING: You are using pip version 19.1.1, however version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
deppe@bardiche:~/nn_dataflow$ python3 -m unittest discover
.............No valid schedule found for AlexNet.
No valid schedule found for AlexNet.
............................................................................SSM1
SSM2
SSM3
..................................................................................FFFF.......................................................................................................................................................................................................................................................................................................................................................................
======================================================================
FAIL: test_3d_mem (nn_dataflow.tests.tool_test.test_nn_dataflow_search.TestNNDataflowSearch)
With 3D memory.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/deppe/nn_dataflow/nn_dataflow/tests/tool_test/test_nn_dataflow_search.py", line 45, in test_3d_mem
    self.assertEqual(ret, 0)
AssertionError: 1 != 0

======================================================================
FAIL: test_default_invoke (nn_dataflow.tests.tool_test.test_nn_dataflow_search.TestNNDataflowSearch)
Default invoke.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/deppe/nn_dataflow/nn_dataflow/tests/tool_test/test_nn_dataflow_search.py", line 40, in test_default_invoke
    self.assertEqual(ret, 0)
AssertionError: 1 != 0

======================================================================
FAIL: test_no_dataflow (nn_dataflow.tests.tool_test.test_nn_dataflow_search.TestNNDataflowSearch)
No dataflow scheme found.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/deppe/nn_dataflow/nn_dataflow/tests/tool_test/test_nn_dataflow_search.py", line 53, in test_no_dataflow
    self.assertEqual(ret, 2)
AssertionError: 1 != 2

======================================================================
FAIL: test_default_invoke (nn_dataflow.tests.tool_test.test_nn_layer_stats.TestNNLayerStats)
Default invoke.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/deppe/nn_dataflow/nn_dataflow/tests/tool_test/test_nn_layer_stats.py", line 38, in test_default_invoke
    self.assertEqual(ret, 0)
AssertionError: 1 != 0

----------------------------------------------------------------------
Ran 534 tests in 842.881s

FAILED (failures=4)

```

**Comment by johndeppe** [comment]
Solved it. Ubuntu's use of `python3` was the problem. Changing `python` to `python3` in the invocations in the failing files fixed it.

**Comment by gaomy3832** [comment]
Thanks for reporting this! We missed those places when upgrading to python 3. Now should be fixed with 5ae8eeba4e243df6e9a69127073513a852a62d17
